src/kaleidoscope/plugin/IdleLEDs.cpp

```diff
@@ -25,23 +26,33 @@ uint16_t IdleLEDs::idle_time_limit = 600; // 10 minutes
 uint32_t IdleLEDs::end_time_;
 
 EventHandlerResult IdleLEDs::beforeEachCycle() {
-  if (!::LEDControl.paused &&
-      (Kaleidoscope.millisAtCycleStart() >= end_time_)) {
+  if (Kaleidoscope.millisAtCycleStart() < end_time_)
+    return EventHandlerResult::OK;
+
+  if (::Kaleidoscope.has_leds && !::LEDControl.paused) {
     ::LEDControl.set_all_leds_to(CRGB(0, 0, 0));
     ::LEDControl.syncLeds();
 
     ::LEDControl.paused = true;
   }
+  if (::UnderglowControl.has_leds() && !::UnderglowControl.isPaused()) {
+    ::UnderglowControl.setColor(0, 0, 0);
+    ::UnderglowControl.sync();
+    ::UnderglowControl.pause();
+  }
 
   return EventHandlerResult::OK;
 }
 
 EventHandlerResult IdleLEDs::onKeyswitchEvent(Key &mapped_key, byte row, byte col, uint8_t key_state) {
 
-  if (::LEDControl.paused) {
+  if (Kaleidoscope.has_leds && ::LEDControl.paused) {
     ::LEDControl.paused = false;
     ::LEDControl.refreshAll();
   }
+  if (::UnderglowControl.has_leds() && ::UnderglowControl.isPaused()) {
```

Comment:
has_leds() and isPaused() look different. Feels weird.

Reply:
Good point. Should be `hasLeds()` I think, right?